src/java/org/apache/cassandra/repair/autorepair/AutoRepair.java

```diff
@@ -0,0 +1,401 @@
+                                    repairState.setRepairInProgress(false);
+                                    return;
+                                }
+
+                                if (config.getRepairByKeyspace(repairType))
+                                {
+                                    if (AutoRepairUtils.keyspaceMaxRepairTimeExceeded(repairType, tableStartTime, tablesToBeRepaired.size()))
+                                    {
+                                        repairState.setRepairSkippedTablesCount(repairState.getRepairSkippedTablesCount() + tablesToBeRepaired.size());
+                                        logger.info("Keyspace took too much time to repair hence skipping it {}",
+                                                    keyspaceName);
+                                        break;
+                                    }
+                                }
+                                else
+                                {
+                                    if (AutoRepairUtils.tableMaxRepairTimeExceeded(repairType, tableStartTime))
+                                    {
+                                        repairState.setRepairSkippedTablesCount(repairState.getRepairSkippedTablesCount() + 1);
+                                        logger.info("Table took too much time to repair hence skipping it {}.{}",
+                                                    keyspaceName, tableName);
+                                        break;
+                                    }
+                                }
+                                Token childStartToken = token.left;
+                                Token childEndToken = token.right;
+                                logger.debug("Current Token Left side {}, right side {}", childStartToken
+                                                                                          .toString(), childEndToken.toString());
+
+                                ranges.add(new Range<>(childStartToken, childEndToken));
+                                totalProcessedSubRanges++;
+                                if ((totalProcessedSubRanges % config.getRepairThreads(repairType) == 0) ||
+                                    (totalProcessedSubRanges == totalSubRanges))
+                                {
+                                    RepairRunnable task = repairState.getRepairRunnable(keyspaceName,
+                                                                                        config.getRepairByKeyspace(repairType) ? tablesToBeRepaired : ImmutableList.of(tableName),
+                                                                                        ranges, primaryRangeOnly);
+                                    repairState.resetWaitCondition();
+                                    new Thread(NamedThreadFactory.createAnonymousThread(new FutureTask<>(task, null))).start();
```

Comment:
This fails checkstyles requirements. Should create a single thread pool for this or reuse one of existing ones

Reply:
Incorporated